### PR TITLE
Feature/handling singletons

### PR DESCRIPTION
**Pull request by piiertho**
This adds a method to retrieve singleton pointers from engine.
This adds private to `KtObject` field `isSingleton` to avoid singleton registering in GC.

### CODE REVIEW

**Review by chippmann — APPROVED** [review]

File: kt/godot-library/src/main/kotlin/godot/tests.kt
Comment (raniejade) [inline kt/godot-library/src/main/kotlin/godot/tests.kt:9]: `isSingleton` (and `isRef`) is a property of the class not the instance, so it is quite weird to have it as instance fields. Is there a way we can use the types for this checks?

Comment (piiertho) [reply kt/godot-library/src/main/kotlin/godot/tests.kt:9]: No way to use the type and get performances :/
I think I'll wait for merge of other PR and implement the check with internal method with unifying of godot-library and godot-runtime modules

**Review by CedNaru — APPROVED** [review]

**Review by raniejade — APPROVED** [review]
LGTM, if you add a todo to rework it later.

**Review by chippmann — APPROVED** [review]

**Pull request merged**
